docs/11_search.md

```diff
@@ -0,0 +1,286 @@
+
+```sql
+WorkflowType = "main.Workflow" and CloseStatus != 0 and (StartTime > "2019-06-07T16:46:34-08:00" or CloseTime > "2019-06-07T16:46:34-08:00" order by StartTime desc)
+```
+
+## Memo vs Search Attributes
+
+Cadence offers two methods for creating workflows with key-value pairs: memo and search attributes. Memo can only be provided on workflow start. Also, memo data are not indexed. This means they are not searchable. Memo data are visible when listing workflows using the list APIs. Search attributes data are indexed so you can search workflows by querying on these attributes. However, search attributes require the use of ElasticSearch.
+
+Memo and search attributes are available in the Go client in [StartWorkflowOptions](https://godoc.org/go.uber.org/cadence/internal#StartWorkflowOptions).
+
+```go
+type StartWorkflowOptions struct {
+    // ...
+
+    // Memo - Optional non-indexed info that will be shown in list workflow.
+    Memo map[string]interface{}
+
+    // SearchAttributes - Optional indexed info that can be used in query of List/Scan/Count workflow APIs (only
+    // supported when Cadence server is using ElasticSearch). The key and value type must be registered on Cadence server side.
+    // Use GetSearchAttributes API to get valid key and corresponding value type.
+    SearchAttributes map[string]interface{}
+}
+```
+
+In the Java client, the *WorkflowOptions.Builder* has similar methods for [memo](https://static.javadoc.io/com.uber.cadence/cadence-client/2.6.0/com/uber/cadence/client/WorkflowOptions.Builder.html#setMemo-java.util.Map-) and [search attributes](https://static.javadoc.io/com.uber.cadence/cadence-client/2.6.0/com/uber/cadence/client/WorkflowOptions.Builder.html#setSearchAttributes-java.util.Map-).
+
+Some important distinctions between memo and search attributes:
+
+- Memo can support all data types because it is not indexed. Search attributes only support basic data types (including String, Int, Float, Bool, Datetime) because it is indexed by ElasticSearch.
+- Memo allows whatever keys you put. Search attributes require that keys are whitelisted before using them because ElasticSearch has a limit on indexed keys.
+- Memo doesn't require Cadence clusters to depend on ElasticSearch while search attributes only works with ElasticSearch.
+
+## Search Attributes (Go client usage)
+
+When using the Cadence Go client, provide key-values pairs as SearchAttributes in [StartWorkflowOptions](https://godoc.org/go.uber.org/cadence/internal#StartWorkflowOptions).
+
+SearchAttributes is `map[string]interface{}` where the keys need to be whitelisted so that Cadence knows attribute key name and value type. The value provided in the map must be the same type as registered.
+
+### Whitelisting search attributes
```

Comment:
same comment for "Whitelisting"

To match your other headings, please capitalize Search Attributes